src/za/co/k0ma/playlistWriter/PlaylistWriter.kt

```diff
@@ -36,24 +40,138 @@ class PlaylistWriter(private val files: MutableList<File>, private val outputPat
 					+ "$playlistCounter".format("00") + ".m3u8")
 			
 			outputFile.writeText("$PLAYLIST_HEADER\n")
-			writeEntry(fileList, outputFile)
+			writeEntry(fileList, outputFile, readTags)
 		}
 		
 		println("------ Finished writing files ------");
 	}
 	
-	private fun writeEntry(fileList: List<File>, outputFile: File) {
+	private fun writeEntry(fileList: List<File>, outputFile: File, readTags: Boolean) {
 		fileList.forEach {
-			outputFile.appendText("$ENTRY_START" + createTitle(it) + "\n")
+			outputFile.appendText(ENTRY_START + createTitle(it, readTags) + "\n")
 			outputFile.appendText(it.path + "\n")
 		}
 	}
 	
-	fun createTitle(file: File) = file.nameWithoutExtension
+	private fun createTitle(file: File, readTags: Boolean): String {
+		var title = file.nameWithoutExtension
+		
+		if (!readTags) {
+			return title
+		}
+		
+		val extension = file.extension
+		
+		when (extension.toLowerCase()) {
+			"mp3" -> title = readID3Tag(file)
+		}
+		
+		return title
+	}
+	
+	//TODO: Move tag reading stuff to it's own class
+	fun readID3Tag(file: File): String {
+		var randomAccessFile: RandomAccessFile? = null
+		try {
+			randomAccessFile = RandomAccessFile(file, "r");
```

Comment:
This assignment can safely move to the declaration of `randomAccessFile`, avoiding the need for a type-annotation